From 13377dcf55c1863bd0ba1e11eba398e6ccdfe86f Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Fri, 22 Jan 2016 19:39:55 -0500
Subject: [PATCH] Fix a heap overflow in hdr handling

bug: https://github.com/ImageMagick/ImageMagick/issues/90
bug-ubuntu: https://bugs.launchpad.net/ubuntu/+source/imagemagick/+bug/1537213
origin: upstream, https://github.com/ImageMagick/ImageMagick/commit/14e606db148d6ebcaae20f1e1d6d71903ca4a556
Bug-debian: https://bugs.debian.org/832469
---
 coders/hdr.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: imagemagick-6.6.9.7/coders/hdr.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/hdr.c	2016-11-15 15:16:53.805843125 -0500
+++ imagemagick-6.6.9.7/coders/hdr.c	2016-11-15 15:16:53.805843125 -0500
@@ -727,10 +727,11 @@
   /*
     Write HDR pixels.
   */
-  pixels=(unsigned char *) AcquireQuantumMemory(image->columns,4*
+  pixels=(unsigned char *) AcquireQuantumMemory(image->columns+128,4*
     sizeof(*pixels));
   if (pixels == (unsigned char *) NULL)
     ThrowWriterException(ResourceLimitError,"MemoryAllocationFailed");
+  (void) ResetMagickMemory(pixels,0,4*(image->columns+128)*sizeof(*pixels));
   for (y=0; y < (ssize_t) image->rows; y++)
   {
     p=GetVirtualPixels(image,0,y,image->columns,1,&image->exception);
