From 5a1eb96fee4f3668d3defe51bd6de7ae25884006 Mon Sep 17 00:00:00 2001
From: dirk <dirk@git.imagemagick.org>
Date: Tue, 2 Feb 2016 21:03:24 +0100
Subject: [PATCH] Fix a heap buffer overflow in psd file handling

Bug: https://github.com/ImageMagick/ImageMagick/issues/92.
Origin: https://github.com/ImageMagick/ImageMagick/commit/30eec879c8b446b0ea9a3bb0da1a441cc8482bc4
Bug-ubuntu: https://bugs.launchpad.net/ubuntu/+source/imagemagick/+bug/1537418
Bug-debian: https://bugs.debian.org/832474
---
 coders/psd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: imagemagick-6.6.9.7/coders/psd.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/psd.c	2016-11-15 15:17:00.893929673 -0500
+++ imagemagick-6.6.9.7/coders/psd.c	2016-11-15 15:17:00.889929624 -0500
@@ -332,6 +332,7 @@
     length++;
     for (j=0; j < (ssize_t) length; j++)
     {
+      CheckNumberCompactPixels;
       switch (depth)
       {
         case 1:
@@ -370,7 +371,6 @@
           break;
         }
       }
-      CheckNumberCompactPixels;
       compact_pixels++;
     }
   }
