Backport of:

From 3dcda687d9867c2ba3e541f76470e50f38900c81 Mon Sep 17 00:00:00 2001
From: dirk <dirk@git.imagemagick.org>
Date: Sat, 6 Feb 2016 12:26:49 +0100
Subject: [PATCH] Fixed head buffer overflow in psd file

bug: https://github.com/ImageMagick/ImageMagick/issues/98
origin: upstream, https://github.com/ImageMagick/ImageMagick/commit/5f16640725b1225e6337c62526e6577f0f88edb8
bug-ubuntu: https://bugs.launchpad.net/ubuntu/+source/imagemagick/+bug/1537424
bug-debian: https://bugs.debian.org/832480
---
 coders/psd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: imagemagick-6.6.9.7/coders/psd.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/psd.c	2016-11-16 07:37:46.732455375 -0500
+++ imagemagick-6.6.9.7/coders/psd.c	2016-11-16 07:37:46.728455325 -0500
@@ -680,16 +680,16 @@
     if (image->depth == 1)
       {
         if (image->compression != RLECompression)
-          count=ReadBlob(image,(image->columns+7)/8,pixels);
+          count=ReadBlob(image,((image->columns+7)/8)*packet_size,pixels);
         else
           {
             count=ReadBlob(image,(size_t) offsets[y],compact_pixels);
             if (count != (ssize_t) offsets[y])
               break;
             count=DecodePSDPixels((size_t) offsets[y],compact_pixels,
-              (ssize_t) 123456,(size_t) ((image->columns+7)/8),pixels);
+              (ssize_t) 123456,(size_t) ((image->columns+7)/8)*packet_size,pixels);
           }
-        if (count < (ssize_t) ((image->columns+7)/8))
+        if (count < (ssize_t) (((image->columns+7)/8)*packet_size))
           break;
       }
     else
