Backport of:

From 1d0bc315d5c9af662a78f12fc24f5b12b7728d11 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 24 Jan 2016 17:42:22 -0500
Subject: [PATCH] Fix out of bound for viff file

Bug: https://github.com/ImageMagick/ImageMagick/issues/99
Origin: upstream, https://github.com/ImageMagick/ImageMagick/commit/ca0c886abd6d3ef335eb74150cd23b89ebd17135
bug-ubuntu: https://bugs.launchpad.net/ubuntu/+source/imagemagick/+bug/1537425
bug-debian: https://bugs.debian.org/832483
---
 coders/viff.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

Index: imagemagick-6.6.9.7/coders/viff.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/viff.c	2016-11-15 15:20:30.124484442 -0500
+++ imagemagick-6.6.9.7/coders/viff.c	2016-11-15 15:20:30.120484393 -0500
@@ -62,6 +62,9 @@
 #include "magick/static.h"
 #include "magick/string_.h"
 #include "magick/module.h"
+
+#define MagickMax(X,Y) ((X) > (Y) ? (X) : (Y))
+
 
 /*
   Forward declarations.
@@ -489,8 +492,8 @@
       max_packets=((image->columns+7UL) >> 3UL)*image->rows;
     else
       max_packets=(size_t) (number_pixels*viff_info.number_data_bands);
-    viff_pixels=(unsigned char *) AcquireQuantumMemory(max_packets,
-      bytes_per_pixel*sizeof(*viff_pixels));
+    viff_pixels=(unsigned char *) AcquireQuantumMemory(MagickMax(number_pixels,
+      max_packets),bytes_per_pixel*sizeof(*viff_pixels));
     if (viff_pixels == (unsigned char *) NULL)
       ThrowReaderException(ResourceLimitError,"MemoryAllocationFailed");
     count=ReadBlob(image,bytes_per_pixel*max_packets,viff_pixels);
