From 5975d116f7d5e0ac17e12625229b3cdc3b736ce6 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 8 May 2016 21:50:10 +0200
Subject: [PATCH] Fix an out of bound in xcf file handling

Bug: https://github.com/ImageMagick/ImageMagick/issues/104
Bug: https://github.com/ImageMagick/ImageMagick/issues/103
Bug-ubuntu: https://bugs.launchpad.net/ubuntu/+source/imagemagick/+bug/1539052
Bug-ubuntu: https://bugs.launchpad.net/ubuntu/+source/imagemagick/+bug/1539051
origin: upstream, https://github.com/ImageMagick/ImageMagick/commit/a2e1064f288a353bc5fef7f79ccb7683759e775c
Bug-debian: https://bugs.debian.org/832504
---
 coders/xcf.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

Index: imagemagick-6.6.9.7/coders/xcf.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/xcf.c	2016-11-15 15:20:37.952580025 -0500
+++ imagemagick-6.6.9.7/coders/xcf.c	2016-11-15 15:20:37.948579977 -0500
@@ -58,6 +58,8 @@
 #include "magick/static.h"
 #include "magick/string_.h"
 #include "magick/module.h"
+
+#define MagickMax(X,Y) ((X) > (Y) ? (X) : (Y))
 
 /*
   Typedef declarations.
@@ -354,7 +356,8 @@
     *xcfdata,
     *xcfodata;
 
-  xcfdata=(XCFPixelPacket *) AcquireQuantumMemory(data_length,sizeof(*xcfdata));
+  xcfdata=(XCFPixelPacket *) AcquireQuantumMemory(MagickMax(data_length,
+    tile_image->columns*tile_image->rows),sizeof(*xcfdata));
   if (xcfdata == (XCFPixelPacket *) NULL)
     ThrowBinaryException(ResourceLimitError,"MemoryAllocationFailed",
       image->filename);
