From f2455e2badccac299101751a741ae702bf6796a2 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Fri, 5 Feb 2016 21:20:42 -0500
Subject: [PATCH] Fix a wpg file out of bound

Origin: upstream, https://github.com/ImageMagick/ImageMagick/commit/bef1e4f637d8f665bc133a9c6d30df08d983bc3a
bug: https://github.com/ImageMagick/ImageMagick/issues/120
bug-ubuntu: https://bugs.launchpad.net/ubuntu/+source/imagemagick/+bug/1542114
bug-debian: https://bugs.debian.org/832780
bug: https://github.com/ImageMagick/ImageMagick/issues/120
---
 coders/wpg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: imagemagick-6.6.9.7/coders/wpg.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/wpg.c	2016-11-15 15:26:00.792522000 -0500
+++ imagemagick-6.6.9.7/coders/wpg.c	2016-11-15 15:26:00.788521951 -0500
@@ -1344,7 +1344,7 @@
                   {
                     ldblk=(ssize_t) ((bpp*image->columns+7)/8);
                     BImgBuff=(unsigned char *) AcquireQuantumMemory((size_t)
-                      ldblk,sizeof(*BImgBuff));
+                      ldblk+1,sizeof(*BImgBuff));
                     if (BImgBuff == (unsigned char *) NULL)
                       goto NoMemory;
 
