From 005fd1642984b9356da2ced54c75f71a362b4d2e Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Mon, 7 Mar 2016 07:12:05 -0500
Subject: [PATCH] Fix an out of bound access in pdb file handling

bug: https://github.com/ImageMagick/ImageMagick/issues/143
origin: upstream, https://github.com/ImageMagick/ImageMagick/commit/424d40ebfcde48bb872eba75179d3d73704fdf1f
bug-ubuntu: https://bugs.launchpad.net/ubuntu/+source/imagemagick/+bug/1553366
bug-debian:  https://bugs.debian.org/832791
---
 coders/pdb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: imagemagick-6.6.9.7/coders/pdb.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/pdb.c	2016-11-15 15:27:32.933647071 -0500
+++ imagemagick-6.6.9.7/coders/pdb.c	2016-11-15 15:27:32.929647023 -0500
@@ -405,7 +405,7 @@
       return(GetFirstImageInList(image));
     }
   packets=(bits_per_pixel*image->columns+7)/8;
-  pixels=(unsigned char *) AcquireQuantumMemory(packets+256UL,image->rows*
+  pixels=(unsigned char *) AcquireQuantumMemory(packets+257UL,image->rows*
     sizeof(*pixels));
   if (pixels == (unsigned char *) NULL)
     ThrowReaderException(ResourceLimitError,"MemoryAllocationFailed");
