From de4059c3b4e19998b0e9b0ae5a0331d7a6701bc6 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 12 Mar 2016 08:04:17 -0500
Subject: [PATCH] Fix a SIGABRT in pdb file handling

bug: https://github.com/ImageMagick/ImageMagick/issues/148
bug-ubuntu: https://bugs.launchpad.net/ubuntu/+source/imagemagick/+bug/1556273
origin: upstream, https://github.com/ImageMagick/ImageMagick/commit/53c1dcd34bed85181b901bfce1a2322f85a59472
bug-debian:  https://bugs.debian.org/832793
---
 coders/psd.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

Index: imagemagick-6.6.9.7/coders/psd.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/psd.c	2016-11-16 07:38:52.513258579 -0500
+++ imagemagick-6.6.9.7/coders/psd.c	2016-11-16 07:38:52.509258529 -0500
@@ -2023,9 +2023,12 @@
     p=PushLongPixel(MSBEndian,p,&count);
     if (id == 0x0000040f)
       {
-        (void) CopyMagickMemory(q,q+PSDQuantum(count)+12,length-
-          (PSDQuantum(count)+12)-(q-datum));
-        SetStringInfoLength(bim_profile,length-(PSDQuantum(count)+12));
+        if ((q+PSDQuantum(count)+12) < (datum+length-16))
+          {
+            (void) CopyMagickMemory(q,q+PSDQuantum(count)+12,length-
+              (PSDQuantum(count)+12)-(q-datum));
+            SetStringInfoLength(bim_profile,length-(PSDQuantum(count)+12));
+          }
         break;
       }
     p+=count;
