From abdc631b8d31f3f91121d1c6d4acdd0d9c733e24 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 24 Dec 2016 12:04:40 -0500
Subject: [PATCH] Ipl file missing malloc check

Fix CVE-2016-10144

origin: https://github.com/ImageMagick/ImageMagick/commit/97566cf2806c0a5a86e884c96831a0c3b1ec6c20
bug-debian: https://bugs.debian.org/851485
---
 coders/ipl.c | 2 ++
 1 file changed, 2 insertions(+)

Index: imagemagick-6.6.9.7/coders/ipl.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/ipl.c	2017-03-02 15:11:07.084195019 -0500
+++ imagemagick-6.6.9.7/coders/ipl.c	2017-03-02 15:11:07.080194971 -0500
@@ -542,6 +542,8 @@
   
 
   quantum_info=AcquireQuantumInfo(image_info, image);
+  if (quantum_info == (QuantumInfo *) NULL)
+    ThrowWriterException(ResourceLimitError,"MemoryAllocationFailed");
   if ((quantum_info->format == UndefinedQuantumFormat) &&
       (IsHighDynamicRangeImage(image,&image->exception) != MagickFalse))
     SetQuantumFormat(image,quantum_info,FloatingPointQuantumFormat);
