From c775d0c21f46b47f64f61c82beebb10ba38c6e1e Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Tue, 10 Jan 2017 20:14:38 -0500
Subject: [PATCH] Fix memory leak in MPC file handling

Fix CVE-2017-5507

bug-debian: https://bugs.debian.org/851382
origin: https://github.com/ImageMagick/ImageMagick/commit/4493d9ca1124564da17f9b628ef9d0f1a6be9738
---
 coders/mpc.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

Index: imagemagick-6.6.9.7/coders/mpc.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/mpc.c	2017-03-02 15:18:24.753396651 -0500
+++ imagemagick-6.6.9.7/coders/mpc.c	2017-03-02 15:18:24.749396603 -0500
@@ -66,6 +66,7 @@
 #include "magick/profile.h"
 #include "magick/property.h"
 #include "magick/quantum-private.h"
+#include "magick/resource_.h"
 #include "magick/static.h"
 #include "magick/statistic.h"
 #include "magick/string_.h"
@@ -806,7 +807,9 @@
         /*
           Create image colormap.
         */
-        if (AcquireImageColormap(image,image->colors) == MagickFalse)
+        image->colormap=(PixelPacket *) AcquireQuantumMemory(image->colors+1,
+          sizeof(*image->colormap));
+        if (image->colormap == (PixelPacket *) NULL)
           ThrowReaderException(ResourceLimitError,"MemoryAllocationFailed");
         if (image->colors != 0)
           {
@@ -895,12 +898,9 @@
     if ((image_info->ping != MagickFalse) && (image_info->number_scenes != 0))
       if (image->scene >= (image_info->scene+image_info->number_scenes-1))
         break;
-    status=SetImageExtent(image,image->columns,image->rows);
-    if (status == MagickFalse)
-      {
-        InheritException(exception,&image->exception);
-        return(DestroyImageList(image));
-      }
+    if ((image->columns > (INT_MAX/(5*sizeof(Quantum)))) ||
+        (image->rows > (INT_MAX/(5*sizeof(Quantum)))))
+      ThrowReaderException(ImageError,"WidthOrHeightExceedsLimit");
     /*
       Attach persistent pixel cache.
     */
