From b16ff30d4647a6c8bb99f8653a59a5384a828479 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Thu, 5 Jan 2017 12:09:24 -0500
Subject: [PATCH] Fix Heap-Buffer-Overflow (TIFF)

Fix CVE-2017-5508

bug-debian: https://bugs.debian.org/851381
bug: https://www.imagemagick.org/discourse-server/viewtopic.php?f=3&t=31161
origin: https://github.com/ImageMagick/ImageMagick/commit/379e21cd32483df6e128147af3bc4ce1f82eb9c4
---
 coders/tiff.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

Index: imagemagick-6.6.9.7/coders/tiff.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/tiff.c	2017-03-02 15:18:32.817492482 -0500
+++ imagemagick-6.6.9.7/coders/tiff.c	2017-03-02 15:18:32.813492435 -0500
@@ -1200,8 +1200,9 @@
     if (TIFFIsTiled(tiff) != MagickFalse)
       method=ReadTileMethod;
     quantum_type=RGBQuantum;
-    tiff_pixels=(unsigned char *) AcquireMagickMemory(TIFFScanlineSize(tiff)+
-      sizeof(uint32));
+    tiff_pixels=(unsigned char *) AcquireMagickMemory(MagickMax(
+      TIFFScanlineSize(tiff),(size_t) (image->columns*samples_per_pixel*
+      pow(2.0,ceil(log(bits_per_sample)/log(2.0))))));
     if (tiff_pixels == (unsigned char *) NULL)
       {
         TIFFClose(tiff);
