From 638b1174a65c53e498e0b4bb2f2e59a9590faa3a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bastien=20ROUCARI=C3=88S?= <roucaries.bastien@gmail.com>
Date: Sun, 12 Feb 2017 21:54:25 +0100
Subject: [PATCH] Fix memory corruption heap overflow in psb file

Fix CVE-2017-5510

Bug: https://github.com/ImageMagick/ImageMagick/issues/348
origin: https://github.com/ImageMagick/ImageMagick/commit/e87af64b1ff1635a32d9b6162f1b0e260fb54ed9
bug-debian: https://bugs.debian.org/851376
---
 coders/psd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: imagemagick-6.6.9.7/coders/psd.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/psd.c	2017-03-02 15:19:22.730085625 -0500
+++ imagemagick-6.6.9.7/coders/psd.c	2017-03-02 15:19:22.714085435 -0500
@@ -1784,7 +1784,7 @@
     ? MagickTrue : MagickFalse;
   packet_size=next_image->depth > 8UL ? 2UL : 1UL;
   (void) packet_size;
-  quantum_info=AcquireQuantumInfo(image_info,image);
+  quantum_info=AcquireQuantumInfo(image_info,next_image);
   pixels=GetQuantumPixels(quantum_info);
   for (y=0; y < (ssize_t) next_image->rows; y++)
   {
